pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -112,6 +120,19 @@ type domainContext struct {
 	// cli options
 	versionPtr    *bool
 	hypervisorPtr *string
+	// CPUs management
+	// Array of the information about the resources used by Applications.
+	// Each element corresponds to an Application.
+	resourcesUsedByVMs map[uuid.UUID]ResourcesUsedByVM
+	// Locks access to the resourcesUsedByVMs array
+	vmResourcesLock sync.Mutex
```

Comment:
> Instead of hand-crafted code using a Mutex, how about adding a variant of base.LockedStringMap 

This lock is also used for locking the `hostCpusPinned ` and `hostCpusNum` fields.

> do you actually need to access the ResourcesUsedByVM for uuid=X when processing DomainConfig/DomainStatus for uuid=Y?

Yes, access is necessary. When we add a new VM with pinned CPUs, we 1) assign the CPUs to the VM, and 2) remove the assigned CPUs from the affinity list of the VMs with no CPU pinning enabled. So, it's necessary for step 2.
For example. First, we have one VM running with no CPU pinning enabled. It has an affinity list of 0-15. Then we create a new VM with CPU pinning enabled and 2 CPUs. We now assign CPUs 2-3 to the newly created VM and remove them from the affinity list of the first VM. So the affinity lists look like: VM1: 0,1,4-15, VM2: 2,3

Reply:
FWIW I think it is bad form for the PR submitter to conclude that an issue is resolved; the reveiwer should conclude that and mark it as such.

A detail, but the comment for this field is then incorrect and misleading.

From a code maintenance perspective this looks like "throw in a mutex" as opposed to defining a object with a monitor, and putting that object in a separate package to prevent any other use of the embedded mutex being used. Unfotunately, sprinkled mutexes get abused over time whereas a well-define object which handles concurrent access internally is not subject to the same abuse.

Reply:
Sorry, I didn't see any new comments here on the previous iteration of the review, so I pushed the button. Mea culpa.

So, what would be the suggestion? To create a new structure with the fields:
```
resourcesUsedByVMs map[uuid.UUID]ResourcesUsedByVM
hostCpusPinned map[int]bool
```
 and add under-a-lock functions to access the data there?

Reply:
I think factoring things so that the cpuassignment is done in a separate package where one can do
   cpuassignment.Init(eveCPUs int) - allocates a struct; gets the cpu count from the hypervisor
and then some Allocate(uuid string, pin bool, vCPUs int) and Free(uuid string) functions in that package. Plus the logic that after allocate/free might have an updated shared set; maybe the functions can return a changedShared boolean?  Plus a GetShared() function to get the shared set when updating it?

Then the package can 1) hide the mutex and 2) do invariant checks that each CPU must always be in exactly one set out of (EVE CPUs, assigned to one app instance, shared set). That prevents bugs creeping in over time as others modify the code; if the invariant is violated a log.Fatal is in order since that should never happen.

Reply:
Done. The logic is moved to the CPUAllocator package.